### PR TITLE
Add ability to indicate the group name when adding a user

### PR DESCRIPTION
**Pull request by arichardet**
Can now add a user by specifying either the group name or gid.

### CODE REVIEW

**Comment by arichardet** [comment]
#276

File: resource/user/user.go [hunk 1]
```diff
@@ -93,10 +104,17 @@ func (u *User) Check(resource.Renderer) (resource.TaskStatus, error) {
 
 			switch {
 			case userByName != nil:
-				status.Level = resource.StatusFatal
```
Comment (BrianHicks) [inline resource/user/user.go]: is this case no longer fatal?

Comment (arichardet) [inline resource/user/user.go]: It should not have been fatal. With state=present, if the user exists, we do not change.

**Comment by BrianHicks** [comment]
LGTM on green

**Pull request merged**